pallets/delegation/src/lib.rs

```diff
@@ -306,12 +350,17 @@ pub mod pallet {
 				<ctype::Error<T>>::CTypeNotFound
 			);
 
+			CurrencyOf::<T>::reserve(&payer, deposit_amount)?;
```

Comment:
M(reserve-deposit part 1): I would like to have the reserve call connected to the Deposit amount that we save in the Hierarchy.
At the moment you must always call `reserve` before you call `create_and_store_new_hierarchy`. If you forget to reserve, the deposit and what's actually reserved diverged.